doc/installing-almond-cloud.md

```diff
@@ -0,0 +1,326 @@
+You must an NLP inference server, and you need to
+periodically synchronize your trained models to Thingpedia to reflect any changes
+(for example by polling the [Thingpedia API](/doc/thingpedia-api)).
+It is also recommended to deploy the NLP training server.
+
+The third form is to deploy a **fully custom Thingpedia**. This is **not**
+a recommended configuration, as it is significantly more challenging
+to manage. Furthermore, the new Thingpedia requires you to upload the
+code and credentials (client IDs, API keys) to all services you care about.
+You must also deploy custom NLP models, as the official ones will not
+be compatible. Use this setup only if you absolutely need custom Thingpedia
+interfaces, and cannot provide these interfaces on Thingpedia. This setup
+is also suitable for developing Thingpedia itself. 
+
+### Step 1: Installing web almond and acquiring dependencies
+The code for web almond is located at our [almond-cloud Github repository](https://github.com/stanford-oval/almond-cloud).
+Clone the repository with the following command:
+```bash
+git clone git@github.com:stanford-oval/almond-cloud.git
+```
+
+The code depends on:
+
+- nodejs (>= 8.0)
+- cvc4 (any version, although >= 1.5 is recommended; only the binary is needed, not the library)
+- gm (provided by GraphicsMagic)
+
+Optionally, it depends on:
+
+- libsystemd
+- bubblewrap
+
+These dependencies are used for sandboxing and journal integration.
+
+A working MySQL server is also required. We recommend MariaDB >= 10.2 for best compatibility.
+
+For example, on Ubuntu (>= 18.04):
+```
+sudo apt install nodejs cvc4 graphicsmagick libsystemd-dev bubblewrap -y
+```
+On Fedora:
+```
+sudo dnf install nodejs cvc4 GraphicsMagick systemd-devel bubblewrap -y
+```
+
+
+If you would like to run the MySQL server locally:
+```
+sudo apt install mariadb-server # Ubuntu
+sudo dnf install mariadb-server # Fedora
+```
+
+This repository uses yarn for dependency tracking.
+You should install yarn from [its website](https://yarnpkg.com/en/docs/install).
+And then run:
+
+```
+yarn install
+```
+
+**Note**: due to strict version requirements between the different Almond components, using
+npm is not supported.
+
+Finally, if you want to use custom NLP models, you must install `decanlp`:
+```
+pip3 install --user 'git+https://github.com/stanford-oval/decaNLP.git#egg=decanlp'
+```
+and you must install [almond-tokenizer](https://github.com/stanford-oval/almond-tokenizer).
+
+#### Setting up database encryption
+
+It is *highly recommended* to encrypt your user's data at rest. To do so, configure your Web Almond to link against
+[sqlcipher](https://www.zetetic.net/sqlcipher) instead of sqlite. Place this in your `.yarnrc`:
+
+```
+sqlite "/opt/sqlcipher"
+sqlite_libname sqlcipher
+```
+
+Replace `/opt/sqlcipher` with the actual installation directory of sqlcipher.
+
+Database encryption uses a randomly generated key that is different from each user.
+
+### Step 2: Configuration
+
```

Comment:
Overall good, but I would prefer to have bullet points for this section. This is really long and dense.